src/ssl.ml

```diff
@@ -81,6 +81,24 @@ type bigarray = (char, Bigarray.int8_unsigned_elt, Bigarray.c_layout) Bigarray.A
 
 external get_error_string : unit -> string = "ocaml_ssl_get_error_string"
 
+type err_function = | Get_error | Peek_error | Peek_last_error
+
+external error_struct : err_function -> int * string * string = "ocaml_ssl_error_struct"
```

Comment:
How about we provide a `Ssl.Error` module with `type t = { code: int; lib: string; reason: string }`?

Reply:
can even expose it in the `.mli` as:

```ocaml
module Error : sig
  type t = private { code: int; lib: string; reason: string } 
end
```

to make sure the library is the only one constructing it.